### PR TITLE
compressor/QatAccel: s/boost::optional/std::optional/

### PR DESCRIPTION
**Pull request by tchaikov**
prefer facilities provided by standard library over the ones in
3rd party libraries.

Signed-off-by: Kefu Chai <tchaikov@gmail.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/compressor/QatAccel.h [hunk 1]
```diff
@@ -37,9 +37,9 @@ class QatAccel {
 
   bool init(const std::string &alg);
 
-  int compress(const bufferlist &in, bufferlist &out, boost::optional<int32_t> &compressor_message);
-  int decompress(const bufferlist &in, bufferlist &out, boost::optional<int32_t> compressor_message);
-  int decompress(bufferlist::const_iterator &p, size_t compressed_len, bufferlist &dst, boost::optional<int32_t> compressor_message);
+  int compress(const bufferlist &in, bufferlist &out, std::optional<int32_t> &compressor_message);
+  int decompress(const bufferlist &in, bufferlist &out, std::optional<int32_t> compressor_message);
+  int decompress(bufferlist::const_iterator &p, size_t compressed_len, bufferlist &dst, std::optional<int32_t> compressor_message);
```
Comment (cbodley) [inline src/compressor/QatAccel.h:42]: probably worth updating all of [compressor/Compressor.h](https://github.com/ceph/ceph/blob/217c328/src/compressor/Compressor.h#L93-L97) while we're at it?

**Review by cbodley — APPROVED** [review]
^ but that can always come in a separate PR

File: src/compressor/QatAccel.h (hunk 1 above)
Comment (tchaikov) [reply src/compressor/QatAccel.h:42]: sure. happy to do so.

**Review by cbodley — APPROVED** [review]

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by cbodley** [comment]
passed rgw qa in https://pulpito.ceph.com/cbodley-2022-03-25_20:30:36-rgw-wip-cbodley-testing-distro-default-smithi/

**Pull request merged**